translator/trace/zipkin/status_code.go

```diff
@@ -108,3 +118,40 @@ func attribToStatusCode(v *tracepb.AttributeValue) *int32 {
 	}
 	return nil
 }
+
+const statusCodeUnknown = 2
```

Comment:
Looks like this values is not going to be being used anymore. 

It means that if we get an invalid value in `error` attribute that is out of the scope of `canonicalCodesMap`, we don't use that to identify an error in the span anymore. I'm not sure this is what we need. Why do you think we should change the behavior this way?

And since we have types conversion now, I would consider setting the UnknownError status for `true` boolean value.

IFAIK another case when this value was set in zipkinV2 translation is when `opencensus.status_description` attribute value is provided but `error` attribute is empty. After this change we will be setting SpanStatus = 0 instead of 2 for those spans. That is acceptable change I think.

@bogdandrutu @tigrannajaryan thoughts?

Reply:
> It means that if we get an invalid value in error attribute that is out of the scope of canonicalCodesMap, we don't use that to identify an error in the span anymore. I'm not sure this is what we need. Why do you think we should change the behavior this way?

I pushed a commit to add this back, its not super clean, I'm open to suggestions on how to improve it.


> And since we have types conversion now, I would consider setting the UnknownError status for true boolean value

For the `error` attribute? Which piece of the mapper should I set that on? `fromCensus/fromStatus/fromHTTP` How should that interact with the presence of the other attributes? I've added some test cases, perhaps listing a few ones to add would help clarify for me


> IFAIK another case when this value was set in zipkinV2 translation is when opencensus.status_description attribute value is provided but error attribute is empty. After this change we will be setting SpanStatus = 0 instead of 2 for those spans. That is acceptable change I think.

I actually undid this when I added support for `unknown` back. Let me know what you think the best path forward is

Reply:
> For the error attribute? Which piece of the mapper should I set that on? fromCensus/fromStatus/fromHTTP How should that interact with the presence of the other attributes? I've added some test cases, perhaps listing a few ones to add would help clarify for me

Probably to add another one like `fromErrorTag`.

> I actually undid this when I added support for unknown back. Let me know what you think the best path forward is

ZipkinV1 receiver was defaulting to `0` code if there is message but not error code identified. I think we should use that approach in both v1 and v2 receivers. Setting `Unknown` error code only based on a message doesn't sound right to me.

Reply:
I've rebased and added a commit for these changes